go/cmd/dolt/commands/verify_constraints.go

```diff
@@ -0,0 +1,134 @@
+	ap.ArgListHelp = append(ap.ArgListHelp, [2]string{"table", "The table to check constraints on."})
+	return ap
+}
+
+func (cmd VerifyConstraintsCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv) int {
+	ap := cmd.createArgParser()
+	help, usage := cli.HelpAndUsagePrinters(cli.GetCommandDocumentation(commandStr, verifyConstraintsDocs, ap))
+	apr := cli.ParseArgs(ap, args, help)
+
+	if apr.NArg() == 0 {
+		usage()
+		return 0
+	}
+	tableNames := apr.Args()
+	working, err := dEnv.WorkingRoot(ctx)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.BuildDError("Unable to get working.").AddCause(err).Build(), nil)
+	}
+	fkColl, err := working.GetForeignKeyCollection(ctx)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.BuildDError("Unable to get foreign keys.").AddCause(err).Build(), nil)
+	}
+
+	var accumulatedConstraintErrors []string
+	for _, givenTableName := range tableNames {
+		tbl, tableName, ok, err := working.GetTableInsensitive(ctx, givenTableName)
+		if err != nil {
+			return HandleVErrAndExitCode(errhand.BuildDError("Unable to get table %s.", givenTableName).AddCause(err).Build(), nil)
+		}
+		if !ok {
+			return HandleVErrAndExitCode(errhand.BuildDError("Table %s does not exist.", givenTableName).Build(), nil)
+		}
+		tblSch, err := tbl.GetSchema(ctx)
+		if err != nil {
+			return HandleVErrAndExitCode(errhand.BuildDError("Unable to get schema for %s.", tableName).AddCause(err).Build(), nil)
+		}
+		fks, _ := fkColl.KeysForTable(tableName)
+
+		for _, fk := range fks {
+			childIdx := tblSch.Indexes().GetByName(fk.TableIndex)
+			childIdxRowData, err := tbl.GetIndexRowData(ctx, fk.TableIndex)
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.BuildDError("Unable to get index data for %s.", fk.TableIndex).AddCause(err).Build(), nil)
+			}
+
+			parentTbl, _, ok, err := working.GetTableInsensitive(ctx, fk.ReferencedTableName)
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.BuildDError("Unable to get table %s.", fk.ReferencedTableName).AddCause(err).Build(), nil)
+			}
+			if !ok {
+				return HandleVErrAndExitCode(errhand.BuildDError("Table %s does not exist.", fk.ReferencedTableName).Build(), nil)
+			}
+			parentTblSch, err := parentTbl.GetSchema(ctx)
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.BuildDError("Unable to get schema for %s.", fk.ReferencedTableName).AddCause(err).Build(), nil)
+			}
+			parentIdx := parentTblSch.Indexes().GetByName(fk.ReferencedTableIndex)
+			parentIdxRowData, err := parentTbl.GetIndexRowData(ctx, fk.ReferencedTableIndex)
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.BuildDError("Unable to get index data for %s.", fk.ReferencedTableIndex).AddCause(err).Build(), nil)
+			}
+
+			err = table.ForeignKeyIsSatisfied(ctx, fk, childIdxRowData, parentIdxRowData, childIdx, parentIdx)
+			if err != nil {
+				accumulatedConstraintErrors = append(accumulatedConstraintErrors, err.Error())
```

Comment:
Does this error message include the name of the table with the failing constraint? If not, add it

Reply:
Yep, errors look like: `foreign key violation on `child1_parent2`.`child1`: `(15292,2)``